s3torchconnector/src/s3torchconnector/_s3bucket_key.py

```diff
@@ -0,0 +1,23 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+class S3BucketKey(object):
```

Comment:
A `dataclass` or `namedtuple` would work better here.

Reply:
Do we really need a dataclass or namedtuple here? Can we just use tuples as (bucket,key)? What is the convention for such simple encapsulations?

Reply:
Intention was to save visibility on what we are storing there. I'm afraid, tuple is not providing that.

Reply:
As far as speed is concerned, `namedtuple` performs the same as regular `tuple`. 
It also provides simple getters for the properties so we can do `bucket_key.key` rather than `bucket_key[1]`.